### PR TITLE
Devel - Completed Wok on both #92 and #100

### PR DESCRIPTION
**Pull request by mgmonteleone**
## Issue #92 

Work for issue #92, added both the get all databases as well as get metrics for database.

Added the database metrics to the Measurements class as well.

Currently only getting all metrics for database is supported. if there is need to pull individual metrics, perhaps we can add that later.

--- 

##  Issue #100
As part of completing #100 needed to refactor quite a bit with measurements, including:

- grabbing the unit description from the API
- Changing the _friendly method to use the unit description to properly render bytes of numbers.
- Moved measurements classes to its own module (`measurements.py`)
- had to refactor tests

### CODE REVIEW

File: atlasapi/measurements.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    def __init__(self, data_list: list):
+        try:
+            self.samples: int = len(clean_list(data_list))
+            self.mean: float = float(mean(clean_list(data_list)))
+            self.min: float = float(min(clean_list(data_list)))
+            self.max: float = float(max(clean_list(data_list)))
+        except StatisticsError:
+            logger.warning('Could not compute statistical values.')
+            self.samples: int = 0
+            self.mean: int = 0
```
Comment (dmalik-mdb) [inline atlasapi/measurements.py]: mean as a float for consistency with the declaration above ^^

File: atlasapi/settings.py [hunk 2]
```diff
@@ -156,7 +164,7 @@ class Settings:
 
     # Atlas default pagination
     pageNum = 1
-    itemsPerPage: int  = int(os.getenv('ITEMS_PER_PAGE', 500))
+    itemsPerPage: int = int(os.getenv('ITEMS_PER_PAGE', 500))
     itemsPerPageMin: int = int(os.getenv('ITEMS_PER_PAGE_MIN', 1))
     itemsPerPageMax: int = int(os.getenv('ITEMS_PER_PAGE_MAX', 2000))
```
Comment (dmalik-mdb) [inline atlasapi/settings.py:169]: Not part of this change but I believe generally we want the max to default to 500, that seems to be the max supported by the clusters and projects endpoints: https://www.mongodb.com/docs/atlas/reference/api/clusters-get-all/

File: atlasapi/specs.py [hunk 3]
```diff
@@ -622,6 +251,172 @@ def add_log_file(self, name: AtlasLogNames, file: BinaryIO) -> None:
...
+            period (Optional[AtlasPeriods]):The period for the disk measurements
+            granularity (Optional[AtlasGranularitues]): The granularity for the disk measurements.
+            atlas_obj (atlasapi.atlas.Atlas): A configured Atlas instance to connect to the API with.
+
+        Returns:
+            List[str]: A list of partition names.
+        """
+        if period is None:
+            period = AtlasPeriods.WEEKS_1
+        logger.info(f'The granularity is {granularity}')
```
Comment (dmalik-mdb) [inline atlasapi/specs.py]: fix message to log period rather than granularity

File: atlasapi/specs.py [hunk 4]
```diff
@@ -622,6 +251,172 @@ def add_log_file(self, name: AtlasLogNames, file: BinaryIO) -> None:
...
+            period (Optional[AtlasPeriods]):The period for the disk measurements
+            granularity (Optional[AtlasGranularitues]): The granularity for the disk measurements.
+            atlas_obj (atlasapi.atlas.Atlas): A configured Atlas instance to connect to the API with.
+
+        Returns:
+           Iterable[Union[AtlasMeasurement, Any]: Yields AtlasMeasirements or the original response.
+        """
+        if period is None:
+            period = AtlasPeriods.WEEKS_1
+        logger.info(f'The granularity is {granularity}')
```
Comment (dmalik-mdb) [inline atlasapi/specs.py]: same as above, change log message to print the period

File: atlasapi/measurements.py [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+                self.min: str = hf.format_size(min(clean_list(data_list)))
+                self.max: str = hf.format_size(max(clean_list(data_list)))
+            elif data_type == 'SCALAR':
+                self.mean: str = hf.format_number(mean(clean_list(data_list)))
+                self.min: str = hf.format_number(min(clean_list(data_list)))
+                self.max: str = hf.format_number(max(clean_list(data_list)))
+            else:
+                self.mean: str = hf.format_number(mean(clean_list(data_list)))
+                self.min: str = hf.format_number(min(clean_list(data_list)))
+                self.max: str = hf.format_number(max(clean_list(data_list)))
```
Comment (dmalik-mdb) [inline atlasapi/measurements.py:53]: are these the only measurement units we'll see? I can't get a definitive answer from the API documentation but Ops Manager docs seem to imply that theres more options: https://www.mongodb.com/docs/ops-manager/current/reference/api/measures/get-measurement-types/

**Review by dmalik-mdb — COMMENTED** [review]
Left some comments on minor fixes. The only major thing is whether there are additional measurement units that might be recorded as SCALAR_PER_SECOND in the friendly output. Other than that LGTM and amazing that you knocked this out in a couple of days!

File: atlasapi/measurements.py (hunk 5 above)
Comment (dmalik-mdb) [reply atlasapi/measurements.py:53]: Nevermind - I totally missed the else there. Should be good to go!

File: atlasapi/specs.py (hunk 3 above)
Comment (mgmonteleone) [reply atlasapi/specs.py]: Thanks! Fixed. Will push new commit.

File: atlasapi/specs.py (hunk 4 above)
Comment (mgmonteleone) [reply atlasapi/specs.py]: Fixed here too, new commit coming!

File: atlasapi/measurements.py (hunk 1 above)
Comment (mgmonteleone) [reply atlasapi/measurements.py]: Fixed! New commit coming.

File: atlasapi/settings.py (hunk 2 above)
Comment (mgmonteleone) [reply atlasapi/settings.py:169]: I have not experimented with this yet, but I think we can live this for now, as we will be refactoring the entire paging subsystem.

**Review by dmalik-mdb — APPROVED** [review]
Looking good! 👍
